### PR TITLE
Fix issue where tags that are suffixed `^{}` is shown

### PR DESCRIPTION
**Pull request by hexium310**
`git ls-remote` displays not only simple tags but also tags, with suffix `^{}`. `^{}` is what make the tag dereference to the non-tag object, so we cannot use it to `git push` and so on.

ref: https://git-scm.com/docs/gitrevisions#Documentation/gitrevisions.txt-emltrevgtemegemv0998em

### CODE REVIEW

File: tests/git.zunit [hunk 1]
```diff
@@ -3549,11 +3549,12 @@
         assert $5 same_as 'git fetch origin '
 
         run cat
-        assert ${#lines} equals 4
+        assert ${#lines} equals 5
         assert ${lines[1]} same_as "${fg[yellow]}another-branch$reset_color  ${fg[green]}branch$reset_color  "
```
Comment (chitoku-k) [inline tests/git.zunit:3553]: Out of curiosity, what are the whitespaces at the end of each line?

Comment (hexium310) [reply tests/git.zunit:3553]: The implementation of `_fzf_complete_tabularize` is bad, and so the whitespaces are inserted at the end when the number of the fields and the number of the its arguments are the same 😢

Comment (chitoku-k) [reply tests/git.zunit:3553]: Heck...

**Review by chitoku-k — APPROVED** [review]
Neat!

**Pull request merged**